main.tf

```
@@ -2538,8 +2538,20 @@ data "aws_iam_policy_document" "fargate_fluentbit" {
       resources = var.fargate_fluentbit.s3_bucket_arns
     }
   }
-}
 
+  dynamic "statement" {
+    for_each = length(lookup(var.fargate_fluentbit, "data_firehose_arns", [])) > 0 ? [1] : []
```

this will most likely fail because we are trying to get the number of ARNs which most likely will be computed values. Lets just use a static boolean conditional check here instead:

```suggestion
    for_each = try(var.fargate_fluentbit.send_to_firehose, false) ? [1] : []
```

---

In your case, you have defined it this way to support multiple S3 buckets, which I have just used as instructed. However, when it comes to Kinesis, we only need one Firehose to send the logs, so we can follow your suggestion for that